components/OAuth2.0/authorization.c

```diff
@@ -0,0 +1,283 @@
+    elapsedTime = elapsedTime / 1000;                                                                               // Convert the elapsed time to seconds                                             
+    if (elapsedTime > (HOUR - 300))
+    {
+        tokenExpired = true;  
+        ESP_LOGW(TAG , "Token is Expired");                                                                 // If the elapsed time is greater than the expiration time, set tokenExpired to true
+    }
+    return tokenExpired;
+}
+
+/**
+ * @brief This function extracts specific parameters from a JSON string and 
+ *         assigns them to corresponding fields in a TokenParam structure.
+ * @param[in] Json The input JSON string.
+ * @param[in] token address to save tokin extracted information.
+ * @return false if fail, true if finish successful.
+ */
+static bool ExtractTokenJson(char *Json, Token_t *token) 
+{
+    const char *TAG = "JsonExTraction";
+    cJSON *J_Token = cJSON_Parse(Json);
+
+    if (J_Token == NULL) 
+    {
+        ESP_LOGE(TAG,"%s",Json);
+        ESP_LOGE(TAG, "Failed to parse JSON\n");
+        return false;
+    }
+
+    cJSON *accessTokenObj = cJSON_GetObjectItem(J_Token, "access_token");
+    if (accessTokenObj != NULL && accessTokenObj->type == cJSON_String) 
+    {
+        strncpy(token->AccessToken, accessTokenObj->valuestring, ACCESS_TOKEN_STR_SIZE - 1);
+        token->AccessToken[ACCESS_TOKEN_STR_SIZE - 1] = '\0';
+    }
+
+    cJSON *tokenTypeObj = cJSON_GetObjectItem(J_Token, "token_type");
+    if (tokenTypeObj != NULL && tokenTypeObj->type == cJSON_String) 
+    {
+        strncpy(token->TokenType, tokenTypeObj->valuestring, TOKEN_TYPE_STR_SIZE - 1);
+        token->TokenType[TOKEN_TYPE_STR_SIZE - 1] = '\0';
+    }
+
+    cJSON *expiresInObj = cJSON_GetObjectItem(J_Token, "expires_in");
+    if (expiresInObj->type == cJSON_Number) 
+    {
+        token->ExpiresInMS = expiresInObj->valueint;
+    }
+
+    cJSON *refreshTokenObj = cJSON_GetObjectItem(J_Token, "refresh_token");
+    if (refreshTokenObj != NULL && refreshTokenObj->type == cJSON_String) 
+    {
+        strncpy(token->RefreshToken, refreshTokenObj->valuestring, REFRESH_TOKEN_STP_SIZE - 1);
+        token->RefreshToken[REFRESH_TOKEN_STP_SIZE - 1] = '\0';
+    }
+
+    cJSON *scopeObj = cJSON_GetObjectItem(J_Token, "scope");
+    if (scopeObj != NULL && scopeObj->type == cJSON_String) 
+    {
+        strncpy(token->GrantedScope, scopeObj->valuestring, GRANTED_SCOP_STR_SIZE - 1);
+        token->GrantedScope[GRANTED_SCOP_STR_SIZE - 1] = '\0';
+    }
+
+    cJSON_Delete(J_Token);
+    return true;
+}
+
+/**
+ * @brief This function reads refresh token from spiffs and send request for new token
+ * @return True if token received and saved, false for otherwise
+ */
+static bool Oauth_TokenRenew(void)
+{
+    char receivedData[LONG_BUF];
+    ReadTxtFileFromSpiffs(pInterfaceHandler->ConfigAddressInSpiffs, 
```

Comment:
"pInterfaceHandler" this is global variable, please use the Pascal way for naming.